flake.nix

```diff
@@ -0,0 +1,26 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }: rec {
```

Comment:
It looks to me like you don't need the `rec` here, and several folks [recommend avoiding it](https://nix.dev/anti-patterns/language#rec-expression) when possible.